scripts/linters/custom_eslint_checks/rules/priv.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Add lint check to ensure that all private functions
+ *      /variables names
+ *      start with _,and that all functions/variables that
+ *      start with _ are tagged as private.
+ */
+
+'use strict';
+
+// TODO(#10479): Implement this rule using the nodes instead of tokens.
+module.exports = {
+  meta: {
+    type: 'layout',
+    docs: {
+      description: (
+        'Add lint check to ensure that all private functions/' +
+        'variables names start with _, and that all functions/variables' +
+        'that start with _ are tagged as private'),
+      category: 'Issue',
+      recommended: true
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      expectedAfter: 'all private functions /' +
+      'variables names should start with _'
+    }
+  },
+  create: function(context) {
+    const sourceCode = context.getSourceCode();
+    const lines = sourceCode.lines;
+
+    return {
+      Program: function(node) {
```

Comment:
I think you are doing it in the wrong way. Try to use node wherever possible. Just access variable names and function declaration using nodes.

Reply:
Okay. How do I check if the variable starting with _ is private or not, 
and then, once I do that I will have to return a message saying, variables and functions starting with _ should be private, that's what I have to do right?

Reply:
Sorry for the late reply. Yes, you are right, It's more around Es6 i.e in class. You can say `private X = 5` for a class property and every private function or variable name should start with _ (underscore)

Reply:
Yes, I got that.
You asked me to used the object being passed to check for it being private.
How do I do that?

Do I direct check if object contains string "private" and then, if the line has _ in it?
Eg: I scan and find, private X = 5, contains private and then check if it has _, this one doesn't so, It'll throw a linting error.

Please confirm if I am approaching it right.
@Hudda

Reply:
Yes, you are right. There will be two cases to check:

1. If variable/function prefixed with `private` then check if the variable/function name starts with an underscore.
2. If the variable/function name starts with an underscore then check if it has `private` as a prefix.